servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -173,27 +155,20 @@ public void accessBlock(long sessionId, long blockId) throws NotFoundException {
    * This method is normally triggered from {@link tachyon.client.file.FileOutStream#close()} if and
    * only if {@link tachyon.client.WriteType#isThrough()} is true. The current implementation of
    * checkpointing is that through {@link tachyon.client.WriteType} operations write to
-   * {@link tachyon.underfs.UnderFileSystem} on the client's write path, but under a session temp
-   * directory (temp directory is defined in the worker as {@link #getSessionUfsTmpFolder(long)}).
+   * {@link tachyon.underfs.UnderFileSystem} on the client's write path, but under a temporary file.
    *
-   * @param sessionId The session id of the client who sends the notification
-   * @param fileId The id of the checkpointed file
+   * @param fileId a file id
+   * @param nonce a nonce used for temporary file creation
    * @throws TException if the file does not exist or cannot be renamed
    * @throws IOException if the update to the master fails
    */
-  public void addCheckpoint(long sessionId, long fileId) throws TException, IOException {
-    // TODO(calvin): This part needs to be changed.
-    String srcPath = PathUtils.concatPath(getSessionUfsTmpFolder(sessionId), fileId);
-    String ufsDataFolder = mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER);
+  public void addCheckpoint(long fileId, long nonce) throws TException, IOException {
```

Comment:
then shall we also rename this to `persistFile` to keep consistent with Master side?